### PR TITLE
Check for dataset profile perms before deleting datasets from admin list view

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Check for dataset permissions before deleting data from list view of dataset admin

## Related Issue
https://trello.com/c/nh1PM1ro/869-dataadmin-is-able-to-delete-dataset-without-perms-from-list-view-of-admin-panel

## Checklist

- [ ]  🚀 is the code ready to be merged and go live?
- [ ]  🛠 does it work (build) locally

### Pull Request

- [ ]  📰 good title
- [ ]  📝good description
- [ ]  🔖 issue linked
- [ ]  📖 changelog filled out

### Commits

- [ ]  commits are clean
- [ ]  commit messages are clean

### Code Quality

- [ ]  🚧 no commented out code
- [ ]  🖨 no unnecessary logging
- [ ]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: wazimap_ng/datasets/admin/base_admin_model.py [hunk 1]
```diff
@@ -24,25 +26,39 @@ def delete_selected_data(modeladmin, request, queryset):
...
-                "count": n, "items": model_ngettext(modeladmin.opts, n)
-            }, messages.SUCCESS)
-        # Return None to display the change list page again.
+        user_profiles = permissions.get_objects_for_user(
+            request.user, Profile, include_public=False
+        )
+
+        queryset_count = queryset.count()
+        if queryset_count:
+            if queryset.exclude(profile__in=user_profiles).count() == 0:
```
Comment (adieyal) [inline wazimap_ng/datasets/admin/base_admin_model.py:36]: please change this to something like:

```
selected_objects_without_permission = queryset.exclude(profile__in=user_profiles).count() > 0
if selected_objects_without_permission:
    ...
else:
   ...
```
Makes the code more self-documenting

**Review by adieyal — CHANGES_REQUESTED** [review]
Please add a unit test

**Comment by goyal1092** [comment]
@adieyal Ready for review

**Comment by goyal1092** [comment]
closing this PR as We will not allow user to delete dataset
